src/scripts/quests/questTypes/DefeatGymQuest.ts

```diff
@@ -39,10 +39,15 @@ class DefeatGymQuest extends Quest implements QuestInterface {
     }
 
     get description(): string {
+        const leaderName = GymList[this.gymTown].leaderName.replace(/\d/g, '');
+        const townName = this.gymTown.replace(/\d/g, '');
+        const trial = GymList[this.gymTown]?.displayName?.includes('Trial');
         const desc = [];
-        desc.push(`Defeat ${this.gymTown}`);
-        if (!this.gymTown.includes('Elite') && !this.gymTown.includes('Champion')) {
-            desc.push('gym');
+
+        if (this.gymTown.includes('Elite') || this.gymTown.includes('Champion') || this.gymTown.includes('Gym')) {
+            desc.push(`Defeat ${townName}`);
+        } else {
+            desc.push(`Defeat ${leaderName}'s ${trial ? 'trial' : 'gym'} at ${townName}`);
```

Comment:
Can we also use displayName here? Or add an other displayName for this situation?